mysql-exporter/mysql-exporter.libsonnet

```diff
@@ -26,22 +24,35 @@ secrets {
     MYSQL_HOST: $.mysql_fqdn,
   },
 
+  // optionally define additional variables before DATA_SOURCE_NAME
+  mysql_additional_env_list:: [],
+
   mysqld_exporter_container::
     container.new('mysqld-exporter', $.image) +
     container.withPorts(k.core.v1.containerPort.new('http-metrics', 9104)) +
     container.withArgsMixin([
       '--collect.info_schema.innodb_metrics',
     ]) +
     container.withEnvMap($.mysqld_exporter_env) +
+    container.withEnvMixin($.mysql_additional_env_list) +
     // Force DATA_SOURCE_NAME to be declared after the variables it references
     container.withEnvMap({
       DATA_SOURCE_NAME: '$(MYSQL_USER):$(MYSQL_PASSWORD)@tcp($(MYSQL_HOST):3306)/',
     }),
 
-
   mysql_exporter_deployment:
     deployment.new('%s-mysql-exporter' % $._config.deployment_name, 1, [$.mysqld_exporter_container]),
 
   mysql_exporter_deployment_service:
     k.util.serviceFor($.mysql_exporter_deployment),
+
+  withSecretPassword(name, key='password'):: {
```

Comment:
I moved this function into this file because its not being used anywhere else.

The behaviour of the function has also changed. It writes the secret ref to the new list, rather than appending to the container definition.